src/categorical_algebra/DPO.jl

```diff
@@ -0,0 +1,177 @@
+module DPO
+export rewrite, rewrite_match, valid_dpo, dangling_condition, id_condition, pushout_complement
+
+using ..FinSets
+using ..CSets
+using ..Limits
+using ...Theories
+using ...Theories: attr
+
+
+"""
+    l
+  L ← I
+m ↓   ↓k
+  G ← K
+    g
+
+Compute a pushout complement, componentwise in Set. On a formal level:
+For each component, define K = G / m(L/l(I)). There is a natural injection g: K↪G
+For each component, define k as equal to the map l;m (every element in the image in G is also in K).
```

Comment:
Docstring here and elsewhere should be hard-wrapped to 80 chars. Most editors come with or let you set up a keyboard shortcut for this, which is very convenient.